lib/rbac/filterer.rb

```diff
@@ -797,7 +797,9 @@ def get_belongsto_matches(blist, klass)
     def belongsto_association_filtered?(vcmeta, klass)
       if [ExtManagementSystem, Host].any? { |x| vcmeta.kind_of?(x) }
         # Eject early if true
-        return true if associated_belongsto_models.any? { |associated| klass <= associated }
+        return true if associated_belongsto_models.any? do |associated|
+          klass <= associated && vcmeta.respond_to?(associated.base_model.to_s.tableize)
```

Comment:
I already forget how this code works... 😭 

Okay, just so I am clear:  What `belongsto_association_filtered?` does it checks if we should even allow filtering to be applied for the given model, and if not, the filter is skipped entirely.

The bug in question is that the previous code just checked if the current model was a decedent of the base class, and not check if the relationship being filtered on even existed, right?

Reply:
> I already forget how this code works... 😭

yes, same here :D 


> The bug in question is that the previous code just checked if the current model was a decedent of the base class, and not check if the relationship being filtered on even existed, right?

yes exactly, 

it worked before https://github.com/ManageIQ/manageiq/pull/18654/commits/8fbebdbb34461914c82949b10360fc5e4c508e04

because it was only done for VmOrTemplate association and this relation  `vms_and_templates` exists on all types of `Host` and `ExtManagement`,

but as we allowed Container* model that we need to this check existence of this relations.